aspnetcore/mvc/controllers/routing.md

```diff
@@ -70,667 +65,788 @@ public class HomeController : Controller
 
-Using `MapControllerRoute()` (or similar) will also map attribute-routed controllers.
+* Call <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllers*> inside `UseEndpoints` to map [attribute routed](#ar) controllers.
+* Call <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllerRoute*> or <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapAreaControllerRoute*>, to map [conventionally routed](#cr) controllers.
 
 <a name="routing-conventional-ref-label"></a>
+<a name="cr"></a>
 
-## Conventional routing
+## Conventional endpoint routing
 
-The `default` route:
+Conventional endpoint routing is used with controllers and views. The `default` route:
 
-```csharp
-endpoints.MapControllerRoute("default", "{controller=Home}/{action=Index}/{id?}");
-```
+[!code-csharp[](routing/samples/3.x/main/StartupDefaultMVC.cs?name=snippet2)]
 
-is an example of a *conventional routing*. We call this style *conventional routing* because it establishes a *convention* for URL paths:
+is an example of a *conventional routing*. It's called *conventional routing* because it establishes a *convention* for URL paths:
 
-* the first path segment maps to the controller name
+* The first path segment, `{controller=Home}`, maps to the controller name.
+* The second segment, `{action=Index}`, maps to the [action](#action) name.
+* The third segment, `{id?}` is used for an optional `id`. The `?` in `{id?}` makes it optional. `id` is used to map to a model entity.
 
-* the second maps to the action name.
+Using this `default` route, the URL path:
 
-* the third segment is used for an optional `id` used to map to a model entity
+* `/Products/List` maps to the `ProductsController.List` action.
+* `/Blog/Article/17` maps to `BlogController.Article` and typically model binds the `id` parameter to 17.
 
-Using this `default` route, the URL path `/Products/List` maps to the `ProductsController.List` action, and `/Blog/Article/17` maps to `BlogController.Article`. This mapping is based on the controller and action names **only** and isn't based on namespaces, source file locations, or method parameters.
+This mapping:
 
-> [!TIP]
-> Using conventional routing with the default route allows you to build the application quickly without having to come up with a new URL pattern for each action you define. For an application with CRUD style actions, having consistency for the URLs across your controllers can help simplify your code and make your UI more predictable.
+* Is based on the controller and [action](#action) names **only**.
+* Isn't based on namespaces, source file locations, or method parameters.
+* Is different than typical [attribute routing](#ar) which doesn't include the action name in the route.
```

Comment:
I suggest removing this point. There are lots of cases where users choose to include the action name in the route using attribute routing.